lab/lab-guide.md

```diff
@@ -0,0 +1,1032 @@
+
+1. Start a new chat and begin by modifying our original question and asking, `What are the highest cost bikes?`. You will notice that it responds correctly, but it didn't hit the cache and there were tokens consumed.
+
+1. Close the browser and stop the application by entering `CTRL + C` in the terminal.
+
+1. In the **ChatService.cs** file, find the **ChatService()** constructor. Comment out the line parsing the similarity score from our settings and manually adjust the value from `0.99` to `0.8`. Save the file.
+
+    ```csharp
+    //_cacheSimilarityScore = Double.TryParse(cacheSimilarityScore, out _cacheSimilarityScore) ? _cacheSimilarityScore : 0.99;
+    _cacheSimilarityScore = .8;
+    ```
+
+1. Back in our terminal, start the application using **dotnet run**. Use `CTRL + click` to open the .NET Aspire dashboard. Once the dashboard loads, click the link to open our chat application.
+
+    ```bash
+    dotnet run
+    ```
+
+1. Start a new session and ask the same modified question again, `What are the highest cost bikes?`. This time you will get a cache hit with zero tokens and the `Cache Hit: True` tag appended to the response.
+
+1. Next ask the follow up question, `What about the least expensive?`. Here too you will get a cached response.
+
+1. Spend a few minutes trying different sequences of questions (and follow up questions) and then modifying them with different similarity scores. If you want to start over and do multiple tests using the same series of questions you can click on **Clear Cache** in the app to empty the cache.
+
+1. When you are done, close the browser and stop the application by entering `CTRL + C` in the terminal.
+
+## A semantic cache needs to have context
+
+If you haven't noticed by now, the semantic cache in this lab caches within the **context window** for a session. This is different from how traditional caches work. Just as we saw earlier in the lab, **context matters!** Caching a conversation ensures that what gets returned from the cache is contextually correct. If it didn't do this, users would get unexpected, and likely unacceptable responses.
+
+Here is a simple mental exercise for a semantic cache that *does not* cache the context window. If you first ask an LLM, "What is the most expensive bike?", it will respond, then cache that user prompt and completion. If you then ask, "What about the least expensive?", the context window we built earlier will pass the chat history to the LLM and it will correctly respond with cheap bikes. And the cache will cache that individual user prompt and completion as well.
+
+Now, say another user in a different session asked, "What is the most expensive motorcycle?", the LLM will respond with expensive motorcycle options. If that user then asked, "What about the least expensive?", the cache will return a list of regular, inexpensive bikes, which of course is not what the user was looking for when they asked about motorcycles.
```

Comment:
Just a nit but maybe pose a bike related question vs. motorcycle. What is the most expensive bike seat. Whatever is in the product catalog.

Reply:
Yeah I was wondering what to put here, can change this